compiler/rustc_mir_transform/src/coverage/counters.rs

```diff
@@ -169,22 +169,20 @@ impl CoverageCounters {
         self.bcb_counters[bcb].as_ref()
     }
 
-    pub(super) fn take_bcb_counter(&mut self, bcb: BasicCoverageBlock) -> Option<BcbCounter> {
-        self.bcb_counters[bcb].take()
-    }
-
-    pub(super) fn drain_bcb_counters(
-        &mut self,
-    ) -> impl Iterator<Item = (BasicCoverageBlock, BcbCounter)> + '_ {
+    pub(super) fn bcb_node_counters(
+        &self,
+    ) -> impl Iterator<Item = (BasicCoverageBlock, &BcbCounter)> {
         self.bcb_counters
-            .iter_enumerated_mut()
-            .filter_map(|(bcb, counter)| Some((bcb, counter.take()?)))
+            .iter_enumerated()
+            .filter_map(|(bcb, counter_kind)| Some((bcb, counter_kind.as_ref()?)))
     }
 
-    pub(super) fn drain_bcb_edge_counters(
-        &mut self,
-    ) -> impl Iterator<Item = ((BasicCoverageBlock, BasicCoverageBlock), BcbCounter)> + '_ {
-        self.bcb_edge_counters.drain()
+    pub(super) fn bcb_edge_counters(
+        &self,
+    ) -> impl Iterator<Item = ((BasicCoverageBlock, BasicCoverageBlock), &BcbCounter)> {
```

Comment:
```suggestion
    ) -> impl Iterator<Item = (BasicCoverageBlock, BasicCoverageBlock, &BcbCounter)> {
```
?
Should we add have a little doc-comment explaining which is which among the 2 bcbs?

Reply:
The intent here is that it's a stream of key/value pairs, where each key is a directed edge in the BCB graph (represented by a pair of nodes). That's why I used a nested tuple, and didn't think it necessary to describe the order.

But it doesn't make much difference either way, so I can certainly change it.